### DISCUSSION TITLE
Output from READ to MQTT / Logging to MQTT as an alternative to SYSLOG

### DISCUSSION THREAD

**Discussion by Delta1977** [emsesp/EMS-ESP32#2125] [Ideas]
as introduced in https://github.com/emsesp/EMS-ESP32/issues/2116 i can send a read command via MQTT and the output get logged to the LOG but i didnt get it via MQTT back.

my first idea is to complete the feature and send the output from read to a topic via MQTT.

The better solution then i think about an option to send all Logmessages to MQTT as an alternative to the existing syslog option.

**Comment by proddy** [comment]
Sending a lot of data like that via MQTT would crash something somewhere. Sometimes we're sending 200+ bytes a second. This is why we use eventsource (SSE) to stream the data over HTTP, and then syslog which are all built for high volume data.

Could you share a use-case where this using MQTT would be useful? There might be an alternative way.

**Reply by Delta1977** [reply]
My idea was an alternative to Syslog. Actual I use esm in combination with Homeassistant. As i have no syslog instance I could not persist the logoutput from Ems. With log over Mqtt I can catch the logs in HA. 

If it's to much load when logs go over mqtt it's no problem for me. 
Then I can shoot up a syslog target elsewhere.

**Comment by MichaelDvP** [comment]
@Delta1977 
You can use the `system/send` topic and send a raw bus request telegram. The you'll get the requested telegram in mqtt topic `response` ( or  /api/system/response), if you have enabled response topic in mqtt settings.

**Reply by Delta1977** [reply]
I will try it

**Reply by Delta1977** [reply]
I send following to Ems-esp/system:

{"cmd":"read","data":"8 1"}

I see the output in logs but no mqtt response 😥

**Reply by MichaelDvP** [reply]
The `read` command does not give response (and system device can not be read), only the the `send` command with valid hex telegram. 
Try `{"cmd":"send","data":"00 88 01 1B 19"}` to read devcie 8, telegram 1 offset 27, length 25

**Reply by Delta1977** [reply]
that works
